File: server/datasets/python_source-code/Multiple-B1/organic_2.py
# Binary Tree Traversal - organic submission 2
# In-order traversal implemented as a generator using yield from.

class Node:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None


def bst_insert(root, value):
    if root is None:
        return Node(value)
    if value < root.value:
        root.left = bst_insert(root.left, value)
    else:
        root.right = bst_insert(root.right, value)
    return root


def inorder_gen(node):
    if node is not None:
        yield from inorder_gen(node.left)
        yield node.value
        yield from inorder_gen(node.right)


def main():
    values = [50, 30, 70, 20, 40, 60, 80]
    root = None
    for v in values:
        root = bst_insert(root, v)
    print("In-order (generator):", list(inorder_gen(root)))


if __name__ == "__main__":
    main()

File: server/datasets/java_source-code/Multiple-B1/organic_2.java
// Binary Tree Traversal - organic submission 2
// In-order traversal using list collector.

import java.util.ArrayList;
import java.util.List;

class Node {
    int value;
    Node left;
    Node right;

    public Node(int value) {
        this.value = value;
        this.left = null;
        this.right = null;
    }
}

class Solution {
    public static Node bstInsert(Node root, int value) {
        if (root == null) {
            return new Node(value);
        }
        if (value < root.value) {
            root.left = bstInsert(root.left, value);
        } else {
            root.right = bstInsert(root.right, value);
        }
        return root;
    }

    public static void inorderCollect(Node node, List<Integer> out) {
        if (node != null) {
            inorderCollect(node.left, out);
            out.add(node.value);
            inorderCollect(node.right, out);
        }
    }

    public static void main(String[] args) {
        int[] values = {50, 30, 70, 20, 40, 60, 80};
        Node root = null;
        for (int v : values) {
            root = bstInsert(root, v);
        }
        List<Integer> result = new ArrayList<>();
        inorderCollect(root, result);
        System.out.println("In-order (generator): " + result);
    }
}
